### ISSUE TITLE
Resources stuck in failed state

### ISSUE THREAD

**Issue by smousa** [feist/pcs#57]
My cluster resources are stuck in a failed state.  When I do a debug-start things seem to work fine, but the status reflects otherwise.  When I do a pcs resource cleanup or pcs resource failcount reset, they don't seem to do anything useful.  Also, it appears that the constraints seem to have no effect on how and where resources start.
  
OS: CentOS7
pcs: 0.9.137
pacemaker: 1.1.12
corosync: 2.3.4

Any help would be greatly appreciated!

`pcs cluster cib`
```xml
<cib crm_feature_set="3.0.9" validate-with="pacemaker-2.3" epoch="202" num_updates="19" admin_epoch="0" cib-last-written="Thu Apr 23 19:32:53 2015" have-quorum="1" dc-uuid="2">
  <configuration>
    <crm_config>
      <cluster_property_set id="cib-bootstrap-options">
        <nvpair id="cib-bootstrap-options-have-watchdog" name="have-watchdog" value="false"/>
        <nvpair id="cib-bootstrap-options-dc-version" name="dc-version" value="1.1.12-a14efad"/>
        <nvpair id="cib-bootstrap-options-cluster-infrastructure" name="cluster-infrastructure" value="corosync"/>
        <nvpair id="cib-bootstrap-options-cluster-name" name="cluster-name" value="serviced"/>
        <nvpair id="cib-bootstrap-options-stonith-enabled" name="stonith-enabled" value="false"/>
        <nvpair id="cib-bootstrap-options-no-quorum-policy" name="no-quorum-policy" value="ignore"/>
        <nvpair id="cib-bootstrap-options-last-lrm-refresh" name="last-lrm-refresh" value="1429807927"/>
        <nvpair id="cib-bootstrap-options-resource-stickiness" name="resource-stickiness" value="100"/>
      </cluster_property_set>
    </crm_config>
    <nodes>
      <node id="1" uname="ip-10-111-23-197.zenoss.loc"/>
      <node id="2" uname="ip-10-111-23-78.zenoss.loc"/>
    </nodes>
    <resources>
      <group id="svcd-group">
        <primitive class="ocf" id="outbound-ip" provider="heartbeat" type="IPaddr2">
          <instance_attributes id="outbound-ip-instance_attributes">
            <nvpair id="outbound-ip-instance_attributes-ip" name="ip" value="199.166.22.4"/>
            <nvpair id="outbound-ip-instance_attributes-cidr_netmask" name="cidr_netmask" value="24"/>
            <nvpair id="outbound-ip-instance_attributes-nic" name="nic" value="eth0"/>
          </instance_attributes>
          <operations>
            <op id="outbound-ip-start-timeout-20s" interval="0s" name="start" timeout="20s"/>
            <op id="outbound-ip-stop-timeout-20s" interval="0s" name="stop" timeout="20s"/>
            <op id="outbound-ip-monitor-interval-30s" interval="30s" name="monitor"/>
          </operations>
        </primitive>
        <primitive class="ocf" id="svcd-daemon" provider="zenoss" type="serviced">
          <instance_attributes id="svcd-daemon-instance_attributes">
            <nvpair id="svcd-daemon-instance_attributes-ipaddr" name="ipaddr" value="199.166.22.4"/>
          </instance_attributes>
          <operations>
            <op id="svcd-daemon-start-timeout-60" interval="0s" name="start" timeout="60"/>
            <op id="svcd-daemon-stop-timeout-60" interval="0s" name="stop" timeout="60"/>
            <op id="svcd-daemon-monitor-interval-20" interval="20" name="monitor" timeout="30"/>
          </operations>
        </primitive>
      </group>
    </resources>
    <constraints>
      <rsc_colocation id="colocation-svcd-daemon-outbound-ip-INFINITY" rsc="svcd-daemon" score="INFINITY" with-rsc="outbound-ip"/>
    </constraints>
    <rsc_defaults>
      <meta_attributes id="rsc_defaults-options">
        <nvpair id="rsc_defaults-options-resource-stickiness" name="resource-stickiness" value="200"/>
      </meta_attributes>
    </rsc_defaults>
  </configuration>
  <status>
    <node_state id="2" uname="ip-10-111-23-78.zenoss.loc" in_ccm="true" crmd="online" crm-debug-origin="do_update_resource" join="member" expected="member">
      <transient_attributes id="2">
        <instance_attributes id="status-2">
          <nvpair id="status-2-shutdown" name="shutdown" value="0"/>
          <nvpair id="status-2-last-failure-svcd-daemon" name="last-failure-svcd-daemon" value="1429807653"/>
          <nvpair id="status-2-probe_complete" name="probe_complete" value="true"/>
        </instance_attributes>
      </transient_attributes>
      <lrm id="2">
        <lrm_resources>
          <lrm_resource id="svcd-daemon" type="serviced" class="ocf" provider="zenoss">
            <lrm_rsc_op id="svcd-daemon_last_0" operation_key="svcd-daemon_monitor_0" operation="monitor" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="8:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:7;8:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="40" rc-code="7" op-status="0" interval="0" last-run="1429817598" last-rc-change="1429817598" exec-time="23" queue-time="0" op-digest="fd2d25ebb470945a1fe92d4f1b39995c" on_node="ip-10-111-23-78.zenoss.loc"/>
          </lrm_resource>
          <lrm_resource id="outbound-ip" type="IPaddr2" class="ocf" provider="heartbeat">
            <lrm_rsc_op id="outbound-ip_last_failure_0" operation_key="outbound-ip_monitor_0" operation="monitor" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="7:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:0;7:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="36" rc-code="0" op-status="0" interval="0" last-run="1429817598" last-rc-change="1429817598" exec-time="35" queue-time="0" op-digest="622a004bf76f86814c0a46eda903dbbd"/>
            <lrm_rsc_op id="outbound-ip_last_0" operation_key="outbound-ip_monitor_0" operation="monitor" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="7:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:0;7:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="36" rc-code="0" op-status="0" interval="0" last-run="1429817598" last-rc-change="1429817598" exec-time="35" queue-time="0" op-digest="622a004bf76f86814c0a46eda903dbbd" on_node="ip-10-111-23-78.zenoss.loc"/>
            <lrm_rsc_op id="outbound-ip_monitor_30000" operation_key="outbound-ip_monitor_30000" operation="monitor" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="8:37:0:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:0;8:37:0:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="41" rc-code="0" op-status="0" interval="30000" last-rc-change="1429817599" exec-time="34" queue-time="0" op-digest="81b7a08cdc81ad4a965e95c21354bfdd" on_node="ip-10-111-23-78.zenoss.loc"/>
          </lrm_resource>
        </lrm_resources>
      </lrm>
    </node_state>
    <node_state id="1" uname="ip-10-111-23-197.zenoss.loc" crmd="online" crm-debug-origin="do_update_resource" in_ccm="true" join="member" expected="member">
      <transient_attributes id="1">
        <instance_attributes id="status-1">
          <nvpair id="status-1-shutdown" name="shutdown" value="0"/>
          <nvpair id="status-1-last-failure-svcd-daemon" name="last-failure-svcd-daemon" value="1429817599"/>
          <nvpair id="status-1-probe_complete" name="probe_complete" value="true"/>
          <nvpair id="status-1-fail-count-svcd-daemon" name="fail-count-svcd-daemon" value="INFINITY"/>
        </instance_attributes>
      </transient_attributes>
      <lrm id="1">
        <lrm_resources>
          <lrm_resource id="svcd-daemon" type="serviced" class="ocf" provider="zenoss">
            <lrm_rsc_op id="svcd-daemon_last_failure_0" operation_key="svcd-daemon_stop_0" operation="stop" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="1:37:0:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:6;1:37:0:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="37" rc-code="6" op-status="0" interval="0" last-run="1429817599" last-rc-change="1429817599" exec-time="21" queue-time="0" op-digest="fd2d25ebb470945a1fe92d4f1b39995c"/>
            <lrm_rsc_op id="svcd-daemon_last_0" operation_key="svcd-daemon_stop_0" operation="stop" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="1:37:0:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:6;1:37:0:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="37" rc-code="6" op-status="0" interval="0" last-run="1429817599" last-rc-change="1429817599" exec-time="21" queue-time="0" op-digest="fd2d25ebb470945a1fe92d4f1b39995c" on_node="ip-10-111-23-197.zenoss.loc"/>
          </lrm_resource>
          <lrm_resource id="outbound-ip" type="IPaddr2" class="ocf" provider="heartbeat">
            <lrm_rsc_op id="outbound-ip_last_0" operation_key="outbound-ip_monitor_0" operation="monitor" crm-debug-origin="do_update_resource" crm_feature_set="3.0.9" transition-key="4:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" transition-magic="0:7;4:34:7:ae3f1f54-93af-4f51-8f33-2e11ffd980c4" call-id="32" rc-code="7" op-status="0" interval="0" last-run="1429817598" last-rc-change="1429817598" exec-time="33" queue-time="0" op-digest="622a004bf76f86814c0a46eda903dbbd" on_node="ip-10-111-23-197.zenoss.loc"/>
          </lrm_resource>
        </lrm_resources>
      </lrm>
    </node_state>
  </status>
</cib>
```
`pcs status`
```
Cluster name: serviced
Last updated: Thu Apr 23 19:36:38 2015
Last change: Thu Apr 23 19:32:53 2015
Stack: corosync
Current DC: ip-10-111-23-78.zenoss.loc (2) - partition with quorum
Version: 1.1.12-a14efad
2 Nodes configured
2 Resources configured


Online: [ ip-10-111-23-197.zenoss.loc ip-10-111-23-78.zenoss.loc ]

Full list of resources:

 Resource Group: svcd-group
     outbound-ip	(ocf::heartbeat:IPaddr2):	Started ip-10-111-23-78.zenoss.loc 
     svcd-daemon	(ocf::zenoss:serviced):	FAILED ip-10-111-23-197.zenoss.loc (unmanaged) 

Failed actions:
    svcd-daemon_stop_0 on ip-10-111-23-197.zenoss.loc 'not configured' (6): call=37, status=complete, exit-reason='none', last-rc-change='Thu Apr 23 19:33:19 2015', queued=0ms, exec=21ms
    svcd-daemon_stop_0 on ip-10-111-23-197.zenoss.loc 'not configured' (6): call=37, status=complete, exit-reason='none', last-rc-change='Thu Apr 23 19:33:19 2015', queued=0ms, exec=21ms


PCSD Status:
  ip-10-111-23-197.zenoss.loc: Online
  ip-10-111-23-78.zenoss.loc: Online

Daemon Status:
  corosync: active/enabled
  pacemaker: active/enabled
  pcsd: active/enabled
```

**Comment by tomjelinek** [comment]
Based on the return code 6 the svcd-daemon resource is misconfigured, the return code 6 means: The resource’s configuration is invalid. E.g. required parameters are missing. (see http://clusterlabs.org/doc/en-US/Pacemaker/1.1-pcs/html-single/Pacemaker_Explained/index.html#s-ocf-return-codes)

You didn't define any constraints that should have effect on placing your resources on particular nodes. You need to set a location constraint for that. The collocation constraint is pointless as resources within one group are always started on the same node, unless you use debug-start as stated in pcs help. See http://clusterlabs.org/doc/en-US/Pacemaker/1.1-pcs/html-single/Pacemaker_Explained/index.html#group-resources for more information about resource groups and http://clusterlabs.org/doc/en-US/Pacemaker/1.1-pcs/html-single/Pacemaker_Explained/index.html#idm254536793808 for more information about constraints.

**Comment by abds230** [comment]
How to get resolve, i am confused ????

**Comment by tomjelinek** [comment]
@abds230, this issue has been closed more than 2 years ago. I provided hints to solve the issue. There was no further response so I closed the issue considering it resolved.

I suggest you to open a new issue and start by describing your problem. If the issue is about configuring a cluster rather than using pcs itself, [users@clusterlabs.org mailinglist](http://lists.clusterlabs.org/mailman/listinfo/users) is a better place to ask since it gets to a much broader audience there.
